pennylane/ops/op_math/evolve.py

```diff
@@ -0,0 +1,184 @@
+            time-dependent Hamiltonian. Defaults to XXX.
+        time (str, optional): The name of the time-based parameter in the parametrized Hamiltonian.
+            Defaults to "t".
+        do_queue (bool): determines if the scalar product operator will be queued. Default is True.
+        id (str or None): id for the scalar product operator. Default is None.
+    """
+
+    _name = "ParametrizedEvolution"
+    num_wires = AnyWires
+    # pylint: disable=too-many-arguments, super-init-not-called
+    def __init__(
+        self, H: ParametrizedHamiltonian, params: list, t, dt=None, time="t", do_queue=True, id=None
+    ):
+        self.H = H
+        self.time = time
+        self.dt = dt
+        self.h_params = params
+        self.t = [0, t] if qml.math.size(t) == 1 else t
+        super().__init__(wires=H.wires, do_queue=do_queue, id=id)
+
+    # pylint: disable=arguments-renamed, invalid-overridden-method
+    @property
+    def has_matrix(self):
+        return all(op.has_matrix or isinstance(op, qml.Hamiltonian) for op in self.H.ops)
+
+    # pylint: disable=import-outside-toplevel
+    def matrix(self, wire_order=None):
+        try:
+            import jax.numpy as jnp
+            from jax.experimental.ode import odeint
+        except ImportError as e:
+            raise ImportError(
+                "Module jax is required for ``ParametrizedEvolution`` class. "
+                "You can install jax via: pip install jax"
+            ) from e
+        y0 = jnp.eye(2 ** len(self.wires), dtype=complex)
+
+        def fun(y, t, params):
+            """dy/dt = -i H(t) y"""
+            kwargs = {self.time: t}
+            return -1j * qml.matrix(self.H(params, **kwargs)) @ y
+
+        if self.dt is None:
+            # TODO: Figure out what is 'p', and best values for 'rtol' and 'atol'
+            self.dt = guess_dt(fun, self.t[0], y0, 2, *self.h_params)
+
+        t = jnp.arange(self.t[0], self.t[1], self.dt, dtype=float)
+        result = odeint(fun, y0, t, self.h_params)
+        mat = result[-1]
+        return qml.math.expand_matrix(mat, wires=self.wires, wire_order=wire_order)
+
+
+def guess_dt(fun, t0, y0, p, *args, rtol=1e-4, atol=1e-4):
+    """Compute a guess for the time step.
+
+    This algorithm is described in further detail in:
+
+    `E. Hairer, S. P. Norsett G. Wanner, Solving Ordinary Differential Equations I: Nonstiff Problems, Sec. II.4. <http://mezbanhabibi.ir/wp-content/uploads/2020/01/ordinary-differential-equations-vol.1.-Nonstiff-problems.pdf>`_
+
+    Args:
+        fun (Callable): function to evaluate the time derivative of the solution `y` at time
+            `t` as `func(y, t, *args)`, producing the same shape/structure as `y0`.
+        t0 (float): initial time
+        y0 (ndarray): array or pytree of arrays representing the initial value for the state.
+        p (float): order (still don't know what this is)
```

Comment:
Do we know what it is now? Do we use it anywhere?

Reply:
This is the order of the runge kutta step, you can default it to 4

Reply:
see https://github.com/google/jax/blob/main/jax/experimental/ode.py#L217

Reply:
Nice, thanks!

Reply:
Using 4 as the runge kutta step makes the tests fail (it chooses a too big time-step: 1e-2). I set the default value to 3.

Reply:
NVM, the tests were wrong. I set the default value to 4.